### PR TITLE
Fixes #7: Add support for Terraform 0.12

### LINKED ISSUE #7: Add support for Terraform 0.12

### PR DESCRIPTION
**Pull request by nick4fake**
https://github.com/terraform-google-modules/terraform-google-dataflow/issues/7

### CODE REVIEW

**Comment by nick4fake** [comment]
dataflow-lint
dataflow-tests

File: examples/dlp_api_example/variables.tf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright 2019 Google LLC
+ * Copyright 2018 Google LLC
```
Comment (kopachevsky) [inline examples/dlp_api_example/variables.tf]: why?

Comment (nick4fake) [reply examples/dlp_api_example/variables.tf]: Because linter fails

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,14 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/) and this
 project adheres to [Semantic Versioning](http://semver.org/).
 
+## [Unreleased]
+
+## [2.0.0] - 2019-YeY-ZZ
```
Comment (Jberlinsky) [inline CHANGELOG.md]: Can we remove the timestamp placeholder here?

File: Makefile [hunk 3]
```diff
@@ -91,48 +88,77 @@ version:
...
 docker_run:
 	docker run --rm -it \
-		-e CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE=${CREDENTIALS_PATH} \
+		-e PROJECT_ID \
+		-e REGION \
+		-e ZONE \
+		-e SERVICE_ACCOUNT_EMAIL \
+		-e SERVICE_ACCOUNT_JSON \
 		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
 		-v $(CURDIR):/cft/workdir \
```
Comment (Jberlinsky) [inline Makefile]: Can we quote `$(CURDIR)` while we're at it, per https://github.com/terraform-google-modules/terraform-google-module-template/pull/21?

File: Makefile [hunk 4]
```diff
@@ -91,48 +88,77 @@ version:
...
 docker_create:
 	docker run --rm -it \
-		-e CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE=${CREDENTIALS_PATH} \
+		-e PROJECT_ID \
+		-e REGION \
+		-e ZONE \
+		-e SERVICE_ACCOUNT_EMAIL \
+		-e SERVICE_ACCOUNT_JSON \
 		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
 		-v $(CURDIR):/cft/workdir \
```
Comment (Jberlinsky) [inline Makefile]: Can we quote `$(CURDIR)` while we're at it, per https://github.com/terraform-google-modules/terraform-google-module-template/pull/21?

File: Makefile [hunk 5]
```diff
@@ -91,48 +88,77 @@ version:
...
 docker_converge:
 	docker run --rm -it \
-		-e CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE=${CREDENTIALS_PATH} \
+		-e PROJECT_ID \
+		-e REGION \
+		-e ZONE \
+		-e SERVICE_ACCOUNT_EMAIL \
+		-e SERVICE_ACCOUNT_JSON \
 		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
 		-v $(CURDIR):/cft/workdir \
```
Comment (Jberlinsky) [inline Makefile]: Can we quote `$(CURDIR)` while we're at it, per https://github.com/terraform-google-modules/terraform-google-module-template/pull/21?

File: Makefile [hunk 6]
```diff
@@ -91,48 +88,77 @@ version:
...
 docker_verify:
 	docker run --rm -it \
-		-e CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE=${CREDENTIALS_PATH} \
+		-e PROJECT_ID \
+		-e REGION \
+		-e ZONE \
+		-e SERVICE_ACCOUNT_EMAIL \
+		-e SERVICE_ACCOUNT_JSON \
 		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
 		-v $(CURDIR):/cft/workdir \
```
Comment (Jberlinsky) [inline Makefile]: Can we quote `$(CURDIR)` while we're at it, per https://github.com/terraform-google-modules/terraform-google-module-template/pull/21?

File: Makefile [hunk 7]
```diff
@@ -91,48 +88,77 @@ version:
...
 docker_destroy:
 	docker run --rm -it \
-		-e CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE=${CREDENTIALS_PATH} \
+		-e PROJECT_ID \
+		-e REGION \
+		-e ZONE \
+		-e SERVICE_ACCOUNT_EMAIL \
+		-e SERVICE_ACCOUNT_JSON \
 		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
 		-v $(CURDIR):/cft/workdir \
```
Comment (Jberlinsky) [inline Makefile]: Can we quote `$(CURDIR)` while we're at it, per https://github.com/terraform-google-modules/terraform-google-module-template/pull/21?

File: Makefile [hunk 8]
```diff
@@ -91,48 +88,77 @@ version:
...
-	@echo "Running test-kitchen tests in docker"
+test_integration_docker:
+	docker run --rm -it \
+		-e PROJECT_ID \
+		-e REGION \
+		-e ZONE \
+		-e SERVICE_ACCOUNT_EMAIL \
+		-e SERVICE_ACCOUNT_JSON \
+		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
+		-v $(CURDIR):/cft/workdir \
```
Comment (Jberlinsky) [inline Makefile]: Can we quote `$(CURDIR)` while we're at it, per https://github.com/terraform-google-modules/terraform-google-module-template/pull/21?

File: examples/dlp_api_example/main.tf [hunk 9]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright 2019 Google LLC
+ * Copyright 2018 Google LLC
```
Comment (Jberlinsky) [inline examples/dlp_api_example/main.tf]: Why was this reverted to 2018? It should be 2019 throughout the repository.

**Comment by aaron-lane** [comment]
Fixes #9.

File: examples/dlp_api_example/main.tf (hunk 9 above)
Comment (nick4fake) [reply examples/dlp_api_example/main.tf]: Nope, as checks will fail.

Comment (morgante) [reply examples/dlp_api_example/main.tf]: 2018 is fine.

**Pull request merged**